dbms/src/DataTypes/DataTypeString.h

```diff
@@ -64,6 +65,46 @@ class DataTypeString final : public IDataType
     bool isString() const override { return true; }
     bool isCategorial() const override { return true; }
     bool canBeInsideNullable() const override { return true; }
+
+    void enumerateStreams(const StreamCallback & callback, SubstreamPath & path) const override;
+
+    void serializeBinaryBulkWithMultipleStreams(
+        const IColumn & column,
+        const OutputStreamGetter & getter,
+        size_t offset,
+        size_t limit,
+        bool position_independent_encoding,
+        SubstreamPath & path) const override;
+
+    void deserializeBinaryBulkWithMultipleStreams(
+        IColumn & column,
+        const InputStreamGetter & getter,
+        size_t limit,
+        double avg_value_size_hint,
+        bool position_independent_encoding,
+        SubstreamPath & path) const override;
+
+    enum class SerdesFormat
+    {
+        SizePrefix = 0, // Legacy format, corresponding to `LegacyName`
+        SeparateSizeAndChars = 1, // New format, corresponding to `NameV1`
+    };
+
+    inline static constexpr auto DefaultSerdesFormat = SerdesFormat::SizePrefix;
```

Comment:
What about define the default serialize string format by `STORAGE_FORMAT_CURRENT`? https://github.com/JinheLin/tiflash-1/pull/4